catalog/gke/cluster/cluster.yaml

```diff
@@ -0,0 +1,108 @@
+apiVersion: container.cnrm.cloud.google.com/v1beta1
+kind: ContainerCluster
+metadata:
+  name: example-us-east4 # {"$kpt-set":"cluster-name"}
+  namespace: config-control # {"$kpt-set":"platform-namespace"}
+  annotations:
+    cnrm.cloud.google.com/remove-default-node-pool: "true"
+    cnrm.cloud.google.com/project-id: platform-project-id # {"$kpt-set":"platform-project-id"}
+  labels:
+    gke.io/environ: dev # {"$kpt-set":"environ"}
+spec:
+  location: us-east4 # {"$kpt-set":"location"}
+  releaseChannel:
+    channel: REGULAR
+  workloadIdentityConfig:
+    identityNamespace: platform-project-id.svc.id.goog # {"$kpt-set":"identity-namespace"}
+  addonsConfig:
+    # Enable internal and external L4 and L7 load balancers by default.
+    # https://cloud.google.com/kubernetes-engine/docs/concepts/ingress
+    httpLoadBalancing:
+      disabled: false
+    # Enable Network Policy by default.
+    # https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy
+    networkPolicyConfig:
+      disabled: false
+    # Enable NodeLocal DNSCache by default.
+    # Improves DNS speed and scaling.
+    # https://cloud.google.com/kubernetes-engine/docs/how-to/nodelocal-dns-cache
+    dnsCacheConfig:
+      enabled: true
+    # Disable Config Connector by default.
+    # Prefer centralized config management on the admin cluster.
+    configConnectorConfig:
+      enabled: false
+    # Enable Horizontal Pod Autoscaling by default.
+    # https://cloud.google.com/kubernetes-engine/docs/concepts/horizontalpodautoscaler
+    horizontalPodAutoscaling:
+      disabled: false
+    # Enable Compute Engine persistent disk CSI Driver by default.
+    # https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/gce-pd-csi-driver
+    gcePersistentDiskCsiDriverConfig:
+      enabled: true
+    # Disable Cloud Run on Anthos by default.
+    # https://cloud.google.com/kuberun
+    cloudrunConfig:
+      disabled: true
+    # Disable Istio by Default (Beta).
+    # https://cloud.google.com/istio/docs/istio-on-gke/overview
+    istioConfig:
+      disabled: true
+    # Disable KALM by Default (Beta).
+    # https://cloud.google.com/kubernetes-engine/docs/how-to/add-on/application-delivery
+    kalmConfig:
```

Comment:
Should we be explicit about these or inherit defaults from KCC?

Reply:
Given that people are probably going to be copy/pasting these, I tried to be pretty explicit, rather than minimal. But this seems like a convention to decide for all blueprints.

Reply:
We should not attempt to replicate defaults everywhere. If we're just inheriting the default, don't specify.

Reply:
removed